File: gcc/testsuite/gcc.dg/enum1.c
/* { dg-do run } */
/* { dg-options "-std=gnu89" } */

/* In standard C, enumeration constants always have type int.  If they
   are not representables are int, they're ill-formed.  In GNU C, we
   give such ill-formed constructs well-defined meaning.  Make sure it
   works.  */

#include <stdlib.h>

enum foo
{
  foo1   = 0,
  foo2   = 0xffffffffffffffffULL,
  foo3   = 0xf0fffffffffffffeULL
};

int main ()
{
  if (sizeof (enum foo) != sizeof (unsigned long long))
    abort ();
  exit (0);
}

File: gcc/testsuite/rust/compile/torture/enum1.rs
enum Foo {
    A,
    B,
    C(char),
    D { x: i64, y: i64 },
}

fn main() {
    let _a = Foo::A;
    let _b = Foo::B;
    let _c = Foo::C('x');
    let _d = Foo::D { x: 20, y: 80 };
}
